### PR TITLE
Add MySQL support

### PR DESCRIPTION
**Pull request by andreis**

### CODE REVIEW

File: mysql/conex_mysql.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	dbName := c.Database
+	if dbName == "" {
+		dbName = "test"
+	}
+	env = append(env, fmt.Sprintf("MYSQL_DATABASE=%s", dbName))
+
+	return env
+}
+
+// Box returns an echo client connect to an echo container based on your provided tags.
```
Comment (omeid) [inline mysql/conex_mysql.go]: This should be returns a MySQL client

**Comment by omeid** [comment]
Hi @andreis, 

This looks pretty good other than the small documentation comment fix. It will be good to merge once that is fixed up.

File: mysql/conex_mysql.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+	// set up user/password access
+	if c.User == "" {
+		c.User = "root"
+	}
+
+	if c.User == "root" && c.Password == "" {
+		env = append(env, fmt.Sprintf("MYSQL_ALLOW_EMPTY_PASSWORD=yes"))
+	} else {
+		env = append(env, fmt.Sprintf("MYSQL_ROOT_PASSWORD=%s", c.RootPassword))
```
Comment (omeid) [inline mysql/conex_mysql.go]: Shouldn't `c.RootPassword` be `c.Password`?

File: mysql/conex_mysql.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+	err := c.Wait(Port, MySQLUpWaitTime)
+
+	if err != nil {
+		c.Drop()
+		t.Fatal("MySQL failed to start.", err)
+	}
+
+	t.Log("MySQL is now accepting connections")
+	db, err := sql.Connnect("mysql", config.url())
```
Comment (omeid) [inline mysql/conex_mysql.go]: `sql.Connect` is not a method, it is `sql.Open`.

**Comment by andreis** [comment]
@omeid sorry I don't understand, what comment are you referring to? I used the postgres driver as an example btw

**Comment by omeid** [comment]
@andreis Sorry, I forgot to submit my review. Now you should see it.

**Comment by andreis** [comment]
Done

**Comment by omeid** [comment]
Thank you!

Since I don't use MySQL, I think it is important to have at least some basic test so I don't accidentally break it when updating conex.

I have added a [simple test in postgres](https://github.com/omeid/conex/blob/master/postgres/postgres_test.go), now if you could just copy and port it to MySQL _box_, we can merge this in. :)

**Comment by andreis** [comment]
On it. Btw:

https://github.com/omeid/conex/blob/master/postgres/postgres_test.go#L15

**Comment by omeid** [comment]
Thanks for pointing that out, I will fix it up!

**Comment by andreis** [comment]
Added a test that hangs, I think on https://github.com/omeid/conex/blob/590797d90d2c1146d38a2141838d767706cb3d1b/wait.go#L16 Will look at this later

**Comment by omeid** [comment]
Seems like MySQL may need more than 10 seconds to accept connections on travis.

**Comment by andreis** [comment]
I don't think that is it. I changed it to 5 minutes and it still hangs, also MySQL connects immediately on my machine.

**Comment by omeid** [comment]
On my computer (already running a few services) it takes just over 10 seconds, so I need to bump the wait time to 15 seconds and it just works. It is weird that it didn't on Travis. 🤔

**Comment by andreis** [comment]
Great that it works on your machine!

**Comment by omeid** [comment]
I just bumped the wait time to 15 seconds and still doesn't work on Travis. Weird.

**Comment by andreis** [comment]
It doesn't work on my machine either, it hangs on `wait` which is weird since it should timeout. Maybe the problem is there?

**Comment by omeid** [comment]
The problem seems to happen with docker-in-docker, I can reproduce it locally.

**Comment by omeid** [comment]
It seems like with docker-in-docker MySQL only binds to IPv6 and that means you can't connect to the database over the IPv4 address.

**Comment by omeid** [comment]
Hi Andrei,

While we may have to fix this at some point, I think it would nice that in the meantime you could use it without maintaining a fork or breaking the CI, so as with my initial plan, I have created `github.com/conex` org to move driver packages there, I create the MySQL repository and have added you there as a collaborator with write access.

I will move the postgres, redis, and rethinkdb drivers there too and setup conex tests in a way that pulls them all for testing. This way, we can give the people interested in different drivers full control over how they want to maintain it and also keep the dependency light, there is no reason to pull client libraries for rethinkdb, redis, postgres, mysql, and probably more in the future if you just want to setup a docker container.

Let me know what you think!

Cheers

**Comment by omeid** [comment]
I pushed the [postgres](https://github.com/conex/postgres) repository, you should be able to simply copy most everything and travis is also enabled for conex/mysql, so you should be good to go!

**Comment by omeid** [comment]
Just pushed your work to [conex/mysql](https://github.com/conex/mysql), if you found a fix for the test, you can push it there. 

Thanks for your contribution!

**Comment by andreis** [comment]
Sorry didn't have a lot of spare cycles but I do now, I'll take care of it.

**Comment by omeid** [comment]
No problem at all! You have write permission to that repository, so feel free to push fixes anytime. :)

**Pull request closed without merging**